src/agent/agent.c

```diff
@@ -1491,64 +1553,54 @@ bool agent_start(Agent *agent) {
                         internal_agent_vtable,
                         agent);
         if (r < 0) {
-                hirte_log_errorf("Failed to add manager vtable: %s", strerror(-r));
+                hirte_log_errorf("Failed to add agent vtable: %s", strerror(-r));
                 return false;
         }
 
-        _cleanup_sd_bus_message_ sd_bus_message *reg_m = NULL;
+        _cleanup_sd_bus_message_ sd_bus_message *bus_msg = NULL;
+        sd_bus_error error = SD_BUS_ERROR_NULL;
         r = sd_bus_call_method(
                         agent->peer_dbus,
                         HIRTE_DBUS_NAME,
                         INTERNAL_MANAGER_OBJECT_PATH,
                         INTERNAL_MANAGER_INTERFACE,
                         "Register",
                         &error,
-                        &reg_m,
+                        &bus_msg,
                         "s",
                         agent->name);
         if (r < 0) {
-                hirte_log_errorf("Failed to issue method call: %s", error.message);
+                hirte_log_errorf(
+                                "Agent '%s' connection attempt %d: failure",
+                                agent->name,
+                                agent->connection_retry_count);
                 sd_bus_error_free(&error);
                 return false;
         }
 
-        r = sd_bus_message_read(reg_m, "");
+        r = sd_bus_message_read(bus_msg, "");
         if (r < 0) {
                 hirte_log_errorf("Failed to parse response message: %s", strerror(-r));
                 return false;
         }
 
-        hirte_log_infof("Registered to '%s' as '%s'", agent->orch_addr, agent->name);
+        hirte_log_infof("Agent '%s' connection attempt %d: success", agent->name, agent->connection_retry_count);
```

Comment:
@sdunnagan 

e2e testing, should be fixed, since the output of tests changed from:
```
 agent_start	msg="Registered to 'tcp:host=10.1.0.230,port=2020' as 'bar'"
```
to:
```
agent_connect	msg="Agent 'foo' connection attempt 0: success"
```
Please take a look here
https://github.com/containers/hirte/blob/main/.github/workflows/e2e.yaml#L64

https://github.com/containers/hirte/blob/main/.github/workflows/e2e.yaml#L79

Consider replacing it to:
```
grep "Agent 'foo'.* success" foo.log accordingly 
```

Reply:
I think we should propose another e2e test here

1) podman exec -u test-user -it  hirte-mgr pkill hirte

2) podman exec -w /home/test-user -u test-user -it hirte-agent-foo  tail -1  foo.log| grep "Agent 'foo'.* disconnected"

3) podman exec -t  hirte-mgr su - test-user -c  "(/usr/local/bin/hirte -c /hirte/example-hirte.ini > mgr.log 2>&1 &)"

4) podman exec -w /home/test-user -u test-user -it hirte-agent-foo  tail -1  foo.log| grep "Agent 'foo'.* success"